core/src/main/java/org/commonjava/indy/core/content/DefaultDownloadManager.java

```diff
@@ -1031,23 +1031,24 @@ private Boolean doDelete( final Transfer item, final EventMetadata eventMetadata
         try
         {
             Location loc = item.getLocation();
-            boolean resetReadonly = ( !loc.allowsStoring() && isIgnoreReadonly( eventMetadata ) && loc instanceof CacheOnlyLocation );
-            try
+            ConcreteResource resource;
+            boolean resetReadonly = ( !loc.allowsStoring() && isIgnoreReadonly( eventMetadata ) );
+            if ( resetReadonly )
             {
-                if ( resetReadonly )
+                resource = new ConcreteResource( loc, item.getPath() )
                 {
-                    ( (CacheOnlyLocation) loc ).setReadonly( false );
-                }
-                final ConcreteResource resource = new ConcreteResource( loc, item.getPath() );
-                transfers.delete( resource, eventMetadata );
+                    @Override
+                    public boolean allowsDeletion()
```

Comment:
Shouldn't this override allowsStoring instead of allowsDeletion?

Reply:
the subsequence code only checks allowsDeletion. so override this is good enough.

Reply:
Or maybe not, since we are in doDelete, but resetReadonly is computed from loc.allowsStoring(), so it should probably use loc.allowsDeletion() above.

But some of the promotions also failed on storing a file, so it should be handled there too.

Reply:
We don't need to worry about loc. The subsequent code only check the resource.allowDeletion. wrt the other places, let me check...

Reply:
Hi mate, good catch! I push a second commit to fix the other place, the allowStoring.

Reply:
OK, looks good now.